### PR TITLE
Create main.yml

### PR DESCRIPTION
**Pull request by secobau**

### CODE REVIEW

**Comment by secobau** [comment]
You do not understand. It is a script for Github.

**Comment by alejandro-colomar** [comment]
The executable file attribute can be stored in git.  You should store it on git (and therefore github), and don't set it up on a script every time.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      id: test
+      run: |
+        set -x
+        compose=etc/docker/swarm/docker-compose.yaml
+        script=bin/check-service.sh
+        sudo cp -rv ./run/configs /run
+        sed -i s/worker/manager/ $compose
+        docker swarm init
+        docker stack deploy -c $compose nlb
+        chmod +x $script
```
Comment (alejandro-colomar) [inline .github/workflows/main.yml:42]: This line is not needed  if the script is stored in git with the correct file attributes.

**Comment by secobau** [comment]
> The executable file attribute can be stored in git. You should store it on git (and therefore github), and don't set it up on a script every time.

I would do it anyway.

It is a common practice to grant execution permissions to a file in an automated script to avoid any unpredicted failure.

It is the same as re-creating a folder just to be sure that it exists.

I cannot control what happens in the github repository (maybe someone removed the execution permission) but I do control my script.

**Comment by secobau** [comment]
So no changes will be made to the script.

**Pull request merged**